authors/apps/comments/serializers.py

```diff
@@ -0,0 +1,47 @@
+from django.shortcuts import get_object_or_404
+from rest_framework import serializers
+from authors.apps.authentication.models import User
+from authors.apps.profiles.models import Profile
+from authors.apps.profiles.serializers import ProfileSerializer
+from .models import Comment
+
+
+class CommentSerializer(serializers.ModelSerializer):
+    replies = serializers.SerializerMethodField('get_is_parent')
+    commented_by = serializers.ReadOnlyField(source='commented_by.username')
+    parent = serializers.ReadOnlyField(source='commented_by.parent')
+
+    class Meta:
+        model = Comment
+        fields = ('id', 'body',  'commented_by',
+                  'created_at', 'updated_at', 'replies', 'parent')
+
+    def get_is_parent(self, object):
+        if not object.is_parent:
+            return CommentChildSerializer(object.children(), many=True).data
+        return object.parent
+
+    def validate(self, data):
```

Comment:
Refactor the `validate` method, create a helper method to reduce duplication. Line 24 and line 41